0.data_analysis/plate_eda_analyses/nbconverted/single_cell_fs_plate_cell_count_eda.py

```diff
@@ -0,0 +1,394 @@
+
+    if na_rows.empty:
+        print("None of the cells have missing columns")
+
+    else:
+        print(na_rows)
+
+    sns.barplot(x="Number of Cells", y="Number of Missing Columns", data=na_rows)
+
+    plt.xlabel("Number of Cells")
+    plt.ylabel("Number of Missing Columns")
+    plt.title("Number of Cells Missing Columns")
+
+    plt.savefig(plate_name_to_path_map[_plate_name] / f"{_plate_name}_cells_missing_columns.png")
+
+
+# In[12]:
+
+
+def visualize_cell_counts_per_edge_wells(_edge_well_df, _plate_name):
+    """
+    Visualize the number of cells per edge well category and genotype in a bar plot
+
+    Parameters
+    ----------
+    _edge_well_df: Pandas Dataframe
+        Contains the cell count per edge well category and genotype
+
+    _plate_name: String
+        Name of the plate for labeling the data
+    """
+    f, ax = plt.subplots(figsize=(20, 10))
+
+    sns.barplot(data=_edge_well_df, x="Metadata_genotype", y="Metadata_number_of_singlecells", hue="Metadata_Edge_Well")
+
+    plt.ylabel('Well Cell Count')
+    plt.xlabel('Genotype')
+    plt.title('Number of cells per Genotype and Edge Well')
+    plt.savefig(plate_name_to_path_map[_plate_name] / f"{_plate_name}_genotype_cell_counts_per_well_type.png")
+    plt.close()
+
+
+# In[13]:
+
+
+def visualize_cell_count_per_site(_cell_countdf, _plate_name):
+    """
+    Visualize the cell count in a violin plot
+
+    Parameters
+    ----------
+    _cell_countdf: Pandas Dataframe
+        Conatins the cell count data
+
+    _plate_name: String
+        Name of the plate for labeling the data
+    """
+
+    f, ax = plt.subplots(figsize=(20, 10))
+
+    sns.violinplot(x=cell_count_col, y="Metadata_genotype", data=_cell_countdf)
+    plt.ylabel("Genotype")
+    plt.xlabel("Site Cell Count")
+    plt.title("Site Cell Count for each Genotype")
+    plt.savefig(plate_name_to_path_map[_plate_name] / f"{_plate_name}_violin_site_cell_counts.png")
+
+
+# In[14]:
+
+
+def visualize_cell_count_per_edge_well_genotype(_cell_countdf, _plate_name):
+    """
+    Visualize the cell count in a violin plot
+
+    Parameters
+    ----------
+    _cell_countdf: Pandas Dataframe
+        Conatins the cell count data
+
+    _plate_name: String
+        Name of the plate for labeling the data
+    """
+
+    f, ax = plt.subplots(figsize=(20, 10))
+
+    sns.violinplot(x=cell_count_col, y="Metadata_genotype", data=_cell_countdf)
+    plt.ylabel("Genotype")
+    plt.xlabel("Site Cell Count")
+    plt.title("Site Cell Count for each Genotype")
+    plt.savefig(plate_name_to_path_map[_plate_name] / f"{_plate_name}_violin_site_cell_counts.png")
+
+
+# ## Visualize Number of Missing Columns
+
+# In[15]:
+
+
+visualize_missing_columns_to_cells(plate3df, "plate_3")
+visualize_missing_columns_to_cells(plate3df, "plate_3p")
+visualize_missing_columns_to_cells(plate3df, "plate_5")
+
+
+# ### Specify Edge Wells
+
+# In[16]:
+
+
+start_letter = 'B'
+end_letter = 'G'
+
+start_num = 1
+end_num = 12
```

Comment:
```suggestion
start_row = 'B'
end_row = 'G'

start_col = 1
end_col = 12
```
Recommend using relevant terms to the plate (e.g., the letter represents the rows, and numbers represent the columns) as you are discussing locations on the plate.

Reply:
Good idea